# Screen Time, Productivity and Attention Span

## Introduction

This analysis explores the screen time dataset and its relationship
with productivity and attention span. The goal is to answer the following questions:

- Is there a relationship between screen time and attention span?
- Does the type of app used influence productivity?
- How does notification handling affect attention and productivity?
- Are there differences between students and professionals?

**Tools used:** Python · Pandas · Plotly  
**Dataset:** [Screen Time Data — Kaggle](https://www.kaggle.com/datasets/mexwell/screen-time-and-productivity)

The dataset "Screen Time Data: Productivity and Attention Span" (https://www.kaggle.com/datasets/muhammadalirazazaidi/screen-time-data-productivity-and-attention-span) contains 200 records representing participants from different age groups, occupations and usage patterns. Each record includes information about daily screen time, type of apps used, notification handling strategy, work strategy, and self-reported attention and productivity scores.

## 1. Library Imports

In [1]:
import pandas as pd
import numpy as np
import plotly.express as px

## 2. Dataset Loading

The file `data.csv` downloaded from Kaggle is used.
The first rows are displayed below to understand its structure.

In [2]:
# Load the dataset
df = pd.read_csv("data.csv")

# Quick view to verify the data
df.head()

,Unnamed: 0,Age Group,Gender,Education Level,Occupation,Average Screen Time,Device,Screen Activity,App Category,Screen Time Period,Environment,Productivity,Attention Span,Work Strategy,Notification Handling,Usage of Productivity Apps
0,0,18–24,Male,Undergraduate,Student,More than 10,Smartphone,"Entertainment (gaming, streaming, social media...","Social Media (e.g., Facebook, Instagram, Linke...",Evening (6 PM–10 PM),Quite workplace,Moderately productive,10–30 minutes,Take regular breaks,Check them briefly and resume my work,"Yes, but i did not find them of any help"
1,1,18–24,Male,Undergraduate,Professional,8-10,Smartphone,"Entertainment (gaming, streaming, social media...","Streaming (e.g., YouTube, Netflix)",Late night (10 PM–6 AM),Quite workplace,Moderately productive,More than 1 hour,"None, i prefer to work without any strategies",Ignore them until my task is completed,"No, i do not use them"
2,2,45 and above,Female,Graduate,Professional,4–6,Smartphone,Academic/Work-related,"Social Media (e.g., Facebook, Instagram, Linke...",Afternoon (12 PM–6 PM),I can work in any environment,Moderately productive,10–30 minutes,Take regular breaks,Check them briefly and resume my work,"No, i do not use them"
3,3,25–34,Male,Undergraduate,Professional,8-10,Laptop/PC,Academic/Work-related,"Social Media (e.g., Facebook, Instagram, Linke...",Afternoon (12 PM–6 PM),Quite workplace,"Extremely productive, i efficiently complete m...",10–30 minutes,"None, i prefer to work without any strategies",Ignore them until my task is completed,"No, i do not use them"
4,4,45 and above,Male,Graduate,Professional,8-10,Laptop/PC,Academic/Work-related,"Productivity (e.g., Microsoft Office, Notion)",Afternoon (12 PM–6 PM),I can work in any environment,"Extremely productive, i efficiently complete m...",30–60 minutes,"None, i prefer to work without any strategies",Check them briefly and resume my work,"No, i do not use them"


## 3. Data Preparation

In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 16 columns):
 #   Column                      Non-Null Count  Dtype
---  ------                      --------------  -----
 0   Unnamed: 0                  200 non-null    int64
 1   Age Group                   200 non-null    str  
 2   Gender                      200 non-null    str  
 3   Education Level             200 non-null    str  
 4   Occupation                  200 non-null    str  
 5   Average Screen Time         200 non-null    str  
 6   Device                      200 non-null    str  
 7   Screen Activity             200 non-null    str  
 8   App Category                200 non-null    str  
 9   Screen Time Period          200 non-null    str  
 10  Environment                 198 non-null    str  
 11  Productivity                200 non-null    str  
 12  Attention Span              200 non-null    str  
 13  Work Strategy               197 non-null    str  
 14  Notification Handling

In [4]:
# Count null values per variable in percentages
(df.isnull().mean() * 100).round(2)

Unnamed: 0                    0.0
Age Group                     0.0
Gender                        0.0
Education Level               0.0
Occupation                    0.0
Average Screen Time           0.0
Device                        0.0
Screen Activity               0.0
App Category                  0.0
Screen Time Period            0.0
Environment                   1.0
Productivity                  0.0
Attention Span                0.0
Work Strategy                 1.5
Notification Handling         0.5
Usage of Productivity Apps    0.0
dtype: float64

In [5]:
# An unnecessary column called "Unnamed: 0" is found and removed.
df = df.drop(columns=["Unnamed: 0"])

# Some null values are also found, but since there are few records (200 people),
# this represents: Environment = 1% (2 people), Work Strategy = 1% (2 people).
# These rows are dropped.
df = df.dropna()

In [6]:
# Count null values per variable in percentages after cleaning
(df.isnull().mean() * 100).round(2)

Age Group                     0.0
Gender                        0.0
Education Level               0.0
Occupation                    0.0
Average Screen Time           0.0
Device                        0.0
Screen Activity               0.0
App Category                  0.0
Screen Time Period            0.0
Environment                   0.0
Productivity                  0.0
Attention Span                0.0
Work Strategy                 0.0
Notification Handling         0.0
Usage of Productivity Apps    0.0
dtype: float64

In [7]:
# Check unique values for each column
for col in df.columns:
    print(f"\nColumn: {col}")
    print(df[col].unique())


Column: Age Group
<StringArray>
['18–24', '45 and above', '25–34', 'Below 18', '35–44']
Length: 5, dtype: str

Column: Gender
<StringArray>
['Male', 'Female']
Length: 2, dtype: str

Column: Education Level
<StringArray>
['Undergraduate', 'Graduate', 'High school or below']
Length: 3, dtype: str

Column: Occupation
<StringArray>
['Student', 'Professional']
Length: 2, dtype: str

Column: Average Screen Time
<StringArray>
['More than 10', '8-10', '4–6', '6–8', '2–4', 'Less than 2']
Length: 6, dtype: str

Column: Device
<StringArray>
['Smartphone', 'Laptop/PC', 'Television', 'Tablet']
Length: 4, dtype: str

Column: Screen Activity
<StringArray>
['Entertainment (gaming, streaming, social media, etc.)', 'Academic/Work-related']
Length: 2, dtype: str

Column: App Category
<StringArray>
['Social Media (e.g., Facebook, Instagram, LinkedIn, Twitter)',
                          'Streaming (e.g., YouTube, Netflix)',
               'Productivity (e.g., Microsoft Office, Notion)',
                 

## 4. Variable Transformation

In [8]:
# Average Screen Time: Has values like ['More than 10' '8-10' '4-6' '6-8' '2-4' 'Less than 2']
# Converted to numeric variable based on the average of both values in each range.
screen_map = {
    "Less than 2": 1,
    "2-4": 3,
    "4-6": 5,
    "6-8": 7,
    "8-10": 9,
    "More than 10": 11
}
df["Screen_Time_Hours"] = df["Average Screen Time"].map(screen_map)

In [9]:
# Attention Span: Converted to ordinal numeric variable
# Values: ['10-30 minutes' 'More than 1 hour' '30-60 minutes' 'Less than 10 minutes']
attention_map = {
    "Less than 10 minutes": 1,
    "10-30 minutes": 2,
    "30-60 minutes": 3,
    "More than 1 hour": 4
}
df["Attention_Score"] = df["Attention Span"].map(attention_map)

In [10]:
# Productivity: Converted to ordinal variable
productivity_map = {
    "Unproductive, i might not have completed the task and got carried away": 1,
    "Moderately productive": 2,
    "Extremely productive, i efficiently complete my tasks": 3
}

df["Productivity_Score"] = df["Productivity"].map(productivity_map)

In [11]:
# Age Group: Converted to ordered category
age_order = ["Below 18", "18-24", "25-34", "35-44", "45 and above"]

df["Age Group"] = pd.Categorical(
    df["Age Group"],
    categories=age_order,
    ordered=True
)

C:\Users\Juan\AppData\Local\Temp\ipykernel_13756\2894003319.py:4: Pandas4Warning: Constructing a Categorical with a dtype and values containing non-null entries not in that dtype's categories is deprecated and will raise in a future version.
  df["Age Group"] = pd.Categorical(


In [12]:
# Gender: Converted to simple category
df["Gender"] = df["Gender"].astype("category")

In [13]:
# Education Level: Converted to ordinal variable
edu_order = ["High school or below", "Undergraduate", "Graduate"]

df["Education Level"] = pd.Categorical(
    df["Education Level"],
    categories=edu_order,
    ordered=True
)

In [14]:
# Occupation: Converted to category
df["Occupation"] = df["Occupation"].astype("category")

In [15]:
# Device, Screen Activity, App Category, Screen Time Period: All converted to nominal categories
cols_nominal = [
    "Device",
    "Screen Activity",
    "App Category",
    "Screen Time Period",
    "Environment",
    "Work Strategy",
    "Notification Handling"
]

for col in cols_nominal:
    df[col] = df[col].astype("category")

In [16]:
# Usage of Productivity Apps: Converted to ordinal
apps_map = {
    "No, i do not use them": 0,
    "Yes, but i did not find them of any help": 1,
    "Yes, they are extremely helpful": 2
}

df["Productivity_App_Score"] = df["Usage of Productivity Apps"].map(apps_map)

In [17]:
# Check unique values for each variable again
for col in df.columns:
    print(f"\nColumn: {col}")
    print(df[col].unique())


Column: Age Group
[NaN, '45 and above', 'Below 18']
Categories (5, str): ['Below 18' < '18-24' < '25-34' < '35-44' < '45 and above']

Column: Gender
['Male', 'Female']
Categories (2, str): ['Female', 'Male']

Column: Education Level
['Undergraduate', 'Graduate', 'High school or below']
Categories (3, str): ['High school or below' < 'Undergraduate' < 'Graduate']

Column: Occupation
['Student', 'Professional']
Categories (2, str): ['Professional', 'Student']

Column: Average Screen Time
<StringArray>
['More than 10', '8-10', '4–6', '6–8', '2–4', 'Less than 2']
Length: 6, dtype: str

Column: Device
['Smartphone', 'Laptop/PC', 'Television', 'Tablet']
Categories (4, str): ['Laptop/PC', 'Smartphone', 'Tablet', 'Television']

Column: Screen Activity
['Entertainment (gaming, streaming, social medi..., 'Academic/Work-related']
Categories (2, str): ['Academic/Work-related', 'Entertainment (gaming, streaming, social medi...]

Column: App Category
['Social Media (e.g., Facebook, Instagram, Link..

The key variables to work with are:

* Screen_Time_Hours (daily average screen time): 11, 9, 5, 7, 3 or 1
* Attention_Score (attention span): 1 (Less than 10 min), 2 (10-30 min), 3 (30-60 min), 4 (More than 1 hour)
* Productivity_Score: 1 (Unproductive), 2 (Moderately productive), 3 (Extremely productive)

## 5. Charts and Correlations

### 5.1. Screen Time vs Attention Span

In [18]:
df[["Screen_Time_Hours", "Attention_Score"]].corr()

,Screen_Time_Hours,Attention_Score
Screen_Time_Hours,1.000000,0.093761
Attention_Score,0.093761,1.000000


Correlation between Screen Time and Attention Span: 0.096.

Conclusion: The correlation is very weak, positive and practically non-existent. There is no significant linear relationship between the hours spent in front of screens and the self-reported attention span.

### 5.2. Screen Time vs Productivity

In [19]:
df[["Screen_Time_Hours", "Productivity_Score"]].corr()

,Screen_Time_Hours,Productivity_Score
Screen_Time_Hours,1.000000,0.044051
Productivity_Score,0.044051,1.000000


Correlation between Screen Time and Productivity: 0.010.

Conclusion: There is also no correlation between screen time and the person's productivity. The data suggests that the amount of time in front of screens does not linearly determine self-perceived productivity.

### 5.3. Average Screen Time by Age Group

In [20]:
df.groupby("Age Group")["Screen_Time_Hours"].mean()

Age Group
Below 18        8.5
45 and above    9.0
Name: Screen_Time_Hours, dtype: float64

Average screen time by age group:
* Under 18: 5.94 hours
* 18–24: 7.09 hours
* 25–34: 8.00 hours
* 35–44: 7.55 hours
* 45 and above: 6.60 hours

Conclusion: Young adults (25–34) show the highest average daily screen time, while those under 18 show the lowest exposure in this sample.

### 5.4. Screen Usage Period vs Attention Span

In [21]:
df.groupby("Screen Time Period")["Attention_Score"].mean()

Screen Time Period
Afternoon (12 PM–6 PM)     3.142857
Evening (6 PM–10 PM)       2.428571
Late night (10 PM–6 AM)    3.166667
Morning (6 AM–12 PM)       2.000000
Name: Attention_Score, dtype: float64

Average attention span by time of day screen is used:
* Afternoon: 2.64
* Evening: 2.53
* Morning: 2.50
* Night: 2.52

Conclusion: No significant differences in attention span are observed based on the time of day screens are used.

### 5.5. Productivity vs Notification Handling

In [22]:
df.groupby("Notification Handling")["Productivity_Score"].mean()

Notification Handling
Check them briefly and resume my work            2.000000
Ignore them until my task is completed           2.418605
Spend time interacting with the notifications    1.444444
Turn off notifications altogether                2.125000
Name: Productivity_Score, dtype: float64

Productivity according to how the person handles notifications:
* Check briefly and resume: 2.00
* Ignore until task complete: 2.17
* Interact with notifications: 1.84
* Silent but check periodically: 2.05

Conclusion: Those who ignore notifications until completing a task show the highest average productivity.

### 5.6. Work Strategy vs Attention Span

In [23]:
df.groupby("Work Strategy")["Attention_Score"].mean()

Work Strategy
Eliminate all distractions                            2.500000
None, i prefer to work without any strategies         2.968750
Take regular breaks                                   2.448276
Use some productivity tool (e.g, timers, blockers)    4.000000
Name: Attention_Score, dtype: float64

Attention span grouped by work strategy:
* Eliminate distractions: 2.41
* None: 2.79
* Take breaks: 2.36
* Time-blocking: 2.59

Conclusion: No work strategy shows a clearly dominant advantage in terms of attention span.

### 5.7. Barplot: Number of People by Daily Screen Time

In [24]:
import plotly.express as px
import pandas as pd

# Explicit order for categories
age_order = ["Below 18", "18-24", "25-34", "35-44", "45 and above"]
screen_order = ["Less than 2", "2-4", "4-6", "6-8", "8-10", "More than 10"]
period_order = ["Morning (6 AM-12 PM)", "Afternoon (12 PM-6 PM)", "Evening (6 PM-10 PM)", "Night (10 PM-6 AM)"]

In [25]:
counts = (
    df["Average Screen Time"]
    .value_counts(normalize=True)
    .reindex(screen_order)
    .reset_index()
)

counts.columns = ["Average Screen Time", "Percentage"]
counts["Percentage"] = counts["Percentage"] * 100

fig1 = px.bar(
    counts,
    x="Average Screen Time",
    y="Percentage",
    category_orders={"Average Screen Time": screen_order},
    text=counts["Percentage"].round(1).astype(str) + "%",
    title="Distribution of Daily Screen Time",
    color_discrete_sequence=["#1f77b4"]
)

fig1.update_layout(
    template="simple_white",
    xaxis_title="Daily Screen Time",
    yaxis_title="Percentage of Participants (%)"
)

fig1.show()
fig1.write_image("barplot_tiempo_pantalla.png")

![Barplot screen time](barplot_tiempo_pantalla.png)

Conclusion:
* Most participants report between 4 and 8 hours of daily screen use, reflecting high digital exposure
* Very few participants report less than 2 hours or more than 10 hours per day

### 5.8. Boxplot: Screen Time vs Attention Span

In [26]:
fig2 = px.box(
    df,
    x="Average Screen Time",
    y="Attention_Score",
    category_orders={"Average Screen Time": screen_order},
    points="all",
    title="Self-Reported Attention Span by Daily Screen Time",
    color_discrete_sequence=["#1f77b4"]
)

fig2.update_layout(
    template="simple_white",
    xaxis_title="Daily Screen Time",
    yaxis_title="Attention Score"
)

fig2.show()
fig2.write_image("boxplot_atencion_pantalla.png")

![Boxplot attention vs screen time](boxplot_atencion_pantalla.png)

Conclusions:
* Attention medians are quite similar across screen time ranges
* No progressive decline in attention is observed as screen time increases
* The distributions overlap considerably, confirming the low correlation

### 5.9. Boxplot: Screen Time vs Productivity

In [27]:
fig3 = px.box(
    df,
    x="Average Screen Time",
    y="Productivity_Score",
    category_orders={"Average Screen Time": screen_order},
    points="all",
    title="Productivity vs Screen Time",
    color_discrete_sequence=["#1f77b4"]
)

fig3.update_layout(
    template="simple_white",
    xaxis_title="Daily Screen Time",
    yaxis_title="Productivity Score"
)

fig3.show()
fig3.write_image("boxplot_productividad_pantalla.png")

![Boxplot productivity vs screen time](boxplot_productividad_pantalla.png)

As screen time increases:
* No progressive decline in productivity median is observed
* Higher usage groups (8–10h, more than 10h) do not show significantly lower productivity
* This reinforces the idea that screen time alone does not determine productivity

### 5.10. Barplot: Notification Handling vs Productivity

In [28]:
notif_mean = (
    df.groupby("Notification Handling", as_index=False)["Productivity_Score"]
    .mean()
    .sort_values("Productivity_Score")
)

fig4 = px.bar(
    notif_mean,
    x="Notification Handling",
    y="Productivity_Score",
    text=notif_mean["Productivity_Score"].round(2),
    title="Average Productivity by Notification Handling Strategy",
    color_discrete_sequence=["#1f77b4"]
)

fig4.update_layout(
    template="simple_white",
    xaxis_title="Notification Handling",
    yaxis_title="Average Productivity Score"
)

fig4.show()
fig4.write_image("barplot_notificaciones_productividad.png")

![Barplot notifications vs productivity](barplot_notificaciones_productividad.png)

Conclusions:
* The way notifications are handled is associated with clear differences in self-perceived productivity
* The group that 'interacts with notifications' shows the lowest average productivity
* Those who 'ignore until task complete' show the highest productivity

### 5.11. Barplot: Work Strategy vs Attention Span

In [29]:
work_mean = (
    df.groupby("Work Strategy", as_index=False)["Attention_Score"]
    .mean()
    .sort_values("Attention_Score")
)

fig5 = px.bar(
    work_mean,
    x="Work Strategy",
    y="Attention_Score",
    text=work_mean["Attention_Score"].round(2),
    title="Average Attention Span by Work Strategy",
    color_discrete_sequence=["#1f77b4"]
)

fig5.update_layout(
    template="simple_white",
    xaxis_title="Work Strategy",
    yaxis_title="Average Attention Score"
)

fig5.show()
fig5.write_image("barplot_estrategia_atencion.png")

![Barplot work strategy vs attention](barplot_estrategia_atencion.png)

Conclusions:
* The declared work strategy does not seem to generate drastic changes in self-perceived attention
* The 'No strategy' group shows the highest average attention, which may reflect a measurement bias

### 5.12. Barplot: Age Group vs Screen Time

In [30]:
age_mean = (
    df.groupby("Age Group", as_index=False)["Screen_Time_Hours"]
    .mean()
)

fig6 = px.bar(
    age_mean,
    x="Age Group",
    y="Screen_Time_Hours",
    category_orders={"Age Group": age_order},
    text=age_mean["Screen_Time_Hours"].round(2),
    title="Average Daily Screen Time by Age Group",
    color_discrete_sequence=["#1f77b4"]
)

fig6.update_layout(
    template="simple_white",
    xaxis_title="Age Group",
    yaxis_title="Average Screen Time (hours)"
)

fig6.show()
fig6.write_image("barplot_edad_pantalla.png")

![Barplot age vs screen time](barplot_edad_pantalla.png)

Conclusions:
* In this sample, minors do not show the highest level of digital exposure
* The highest consumption is in young adults (25–34) with an average of 8 hours per day
* Exposure decreases progressively from 35 years onwards

### 5.13. Barplot: Notification Handling vs Attention Span

In [31]:
notif_att = (
    df.groupby("Notification Handling", as_index=False)["Attention_Score"]
    .mean()
    .sort_values("Attention_Score")
)

fig7 = px.bar(
    notif_att,
    x="Notification Handling",
    y="Attention_Score",
    text=notif_att["Attention_Score"].round(2),
    title="Average Attention Span by Notification Handling Strategy",
    color_discrete_sequence=["#1f77b4"]
)

fig7.update_layout(
    template="simple_white",
    xaxis_title="Notification Handling",
    yaxis_title="Average Attention Score"
)

fig7.show()
fig7.write_image("barplot_notificaciones_atencion.png")

![Barplot notifications vs attention](barplot_notificaciones_atencion.png)

Conclusions:
* As the strategy better protects attention focus, average attention increases
* The group that stops to interact with notifications shows lower attention than those who ignore them

In [32]:
notif_order = (
    df.groupby("Notification Handling")["Attention_Score"]
    .mean()
    .sort_values()
    .index
    .tolist()
)

fig = px.box(
    df,
    x="Notification Handling",
    y="Attention_Score",
    category_orders={"Notification Handling": notif_order},
    points="all",
    title="Attention Span Distribution by Notification Handling Strategy",
    color_discrete_sequence=["#1f77b4"]
)

fig.update_layout(
    template="simple_white",
    xaxis_title="Notification Handling",
    yaxis_title="Attention Score"
)

fig.show()
fig.write_image("boxplot_notificaciones_atencion.png")

![Boxplot notifications vs attention](boxplot_notificaciones_atencion.png)

### 5.13. Heatmap: Screen Usage Period vs Notification Handling (Productivity as color variable)

In [33]:
pivot = df.pivot_table(
    index="Screen Time Period",
    columns="Notification Handling",
    values="Productivity_Score",
    aggfunc="mean"
)

fig = px.imshow(
    pivot,
    text_auto=True,
    aspect="auto",
    color_continuous_scale="RdYlGn",
    title="Average Productivity by Screen Usage Period and Notification Handling"
)

fig.update_layout(
    xaxis_title="Notification Handling",
    yaxis_title="Screen Usage Period"
)

fig.show()
fig.write_image("heatmap_horario_notificaciones.png")

![Heatmap usage period vs notifications](heatmap_horario_notificaciones.png)

Y Axis: Usage period
X Axis: Notification handling
Color: Average productivity (1= less, 3= more)

Conclusions:
* The 'Interacts with Notifications' column consistently shows lower productivity across all time periods
* The 'Ignore until task complete' group tends to show higher productivity, especially in the morning and afternoon

### 5.14. Boxplot: Productivity vs Notification Handling

In [34]:
notif_order = (
    df.groupby("Notification Handling")["Productivity_Score"]
    .mean()
    .sort_values()
    .index
    .tolist()
)

fig = px.box(
    df,
    x="Notification Handling",
    y="Productivity_Score",
    category_orders={"Notification Handling": notif_order},
    points="all",
    title="Productivity Distribution by Notification Handling Strategy",
    color_discrete_sequence=["#1f77b4"]
)

fig.update_layout(
    template="simple_white",
    xaxis_title="Notification Handling",
    yaxis_title="Productivity Score"
)

fig.show()
fig.write_image("boxplot_productividad_notificaciones.png")

![Boxplot productivity vs notifications](boxplot_productividad_notificaciones.png)

Conclusions:
* 'Interacts with notifications' shows the worst distribution: actively interacting with notifications is associated with lower productivity and greater variability
* 'Ignore until task complete' shows the best productivity distribution

### 5.15. Entertainment vs Academic Use

In [35]:
df.groupby("Screen Activity")["Productivity_Score"].mean()
df.groupby("Screen Activity")["Attention_Score"].mean()


Screen Activity
Academic/Work-related                                    2.971429
Entertainment (gaming, streaming, social media, etc.)    2.466667
Name: Attention_Score, dtype: float64

Attention span by screen activity:
* Academic/Work-related: 2.65
* Entertainment: 2.47

Conclusion:
* Task-oriented use is associated with slightly higher attention than entertainment-oriented use

### 5.16. App Category vs Attention Span

In [36]:
df.groupby("App Category")["Attention_Score"].mean().sort_values()


App Category
Messaging (e.g., WhatsApp, Messenger)                          2.125000
Social Media (e.g., Facebook, Instagram, LinkedIn, Twitter)    2.378378
Streaming (e.g., YouTube, Netflix)                             2.500000
Gaming                                                         3.250000
Productivity (e.g., Microsoft Office, Notion)                  3.526316
Name: Attention_Score, dtype: float64

Attention span by app category used:
* Social Media: 2.37
* Messaging: 2.40
* Streaming: 2.52
* Productivity: 2.90
* Gaming: 2.48

Conclusion: Productivity apps are associated with higher attention span, while social media shows the lowest average attention.

In [37]:
app_labels = {
    "Social Media (e.g., Facebook, Instagram, LinkedIn, Twitter)": "Social Media",
    "Messaging (e.g., WhatsApp, Messenger)": "Messaging",
    "Streaming (e.g., YouTube, Netflix)": "Streaming",
    "Productivity (e.g., Microsoft Office, Notion)": "Productivity",
    "Gaming": "Gaming"
}

df["App_Label"] = df["App Category"].map(app_labels)

app_att = (
    df.groupby("App_Label", as_index=False)["Attention_Score"]
    .mean()
    .sort_values("Attention_Score")
)

fig = px.bar(
    app_att,
    x="App_Label",
    y="Attention_Score",
    text=app_att["Attention_Score"].round(2),
    title="Average Attention Span by App Category",
    color_discrete_sequence=["#1f77b4"]
)

fig.update_layout(
    template="simple_white",
    xaxis_title="App Category",
    yaxis_title="Average Attention Score"
)

fig.show()
fig.write_image("barplot_app_atencion.png")

![Barplot app category vs attention](barplot_app_atencion.png)

### 5.18. Students vs Professionals

In [38]:
df.groupby("Occupation")["Screen_Time_Hours"].mean()

Occupation
Professional    9.790698
Student         9.533333
Name: Screen_Time_Hours, dtype: float64

In [39]:
df.groupby("Occupation")["Attention_Score"].mean()

Occupation
Professional    2.818182
Student         2.595745
Name: Attention_Score, dtype: float64

In [40]:
df.groupby("Occupation")["Productivity_Score"].mean()

Occupation
Professional    2.297297
Student         1.950820
Name: Productivity_Score, dtype: float64

Screen time by occupation:
* Professional: 8 hours
* Student: 6.44 hours

Average attention by occupation:
* Professional: 2.58
* Student: 2.52

Average productivity by occupation:
* Professional: 2.09
* Student: 2.07

Conclusion: Professionals spend more time in front of screens but show similar attention and productivity levels to students.

### 5.19. Productivity Apps — Do they help?

In [41]:
df.groupby("Usage of Productivity Apps")["Productivity_Score"].mean()


Usage of Productivity Apps
No, i do not use them                       2.066667
Yes, but i did not find them of any help    1.906250
Yes, they are extremely helpful             2.250000
Name: Productivity_Score, dtype: float64

Average productivity depending on productivity app usage:
* Do not use apps: 2.07
* Use them but they don't help: 1.91
* Use them and they help: 2.14

Conclusion: Productivity apps only show a benefit when the user perceives them as helpful. Simply using them does not guarantee better results.

### 5.20. Heavy Users vs Light Users

In [42]:
df["Heavy_User"] = df["Screen_Time_Hours"] >= 8


In [43]:
df.groupby("Heavy_User")["Attention_Score"].mean()

Heavy_User
False    2.595745
True     2.818182
Name: Attention_Score, dtype: float64

In [44]:
df.groupby("Heavy_User")["Productivity_Score"].mean()

Heavy_User
False    2.072581
True     2.097222
Name: Productivity_Score, dtype: float64

Attention span:
* Light users = 2.49
* Heavy users = 2.65

Average productivity:
* Light users = 2.07
* Heavy users = 2.09

Conclusion: Heavy users (8+ hours) do not show significantly lower attention or productivity than light users, reinforcing the idea that screen time quantity alone does not determine these outcomes.

In [45]:
tmp = df.groupby("Heavy_User", as_index=False)[["Attention_Score","Productivity_Score"]].mean()
tmp_melt = tmp.melt(id_vars="Heavy_User", var_name="Metric", value_name="Score")

fig = px.bar(
    tmp_melt,
    x="Metric",
    y="Score",
    color="Heavy_User",
    barmode="group",
    title="Heavy Users vs Light Users: Attention and Productivity",
    labels={"Heavy_User": "Heavy User", "Score": "Average Score", "Metric": "Metric"}
)

fig.show()
fig.write_image("barplot_heavy_users.png")

![Barplot heavy users](barplot_heavy_users.png)

## 6. Final Conclusions

- There is no significant correlation between screen time and attention span or productivity
- **Notification handling** is the factor that most impacts productivity: those who ignore them until completing a task show better performance
- **Productivity apps** do not guarantee better results on their own
- **Professionals** spend more hours in front of screens than students, but both groups show similar levels of attention and productivity
- Screen use oriented toward **academic or work tasks** is associated with higher attention than entertainment use
- **Young adults (25–34)** are the group with the highest digital exposure

---
*Analysis by Juan Bautista Acuña — [GitHub](https://github.com/bautistaacuna)*